test/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAHelloWorldProject.cs

```diff
@@ -1123,7 +1120,129 @@ public void It_publishes_with_implicit_rid_with_rid_specific_properties(string e
+            public static class Program
+            {
+                public static void Main()
+                {
+                    Console.WriteLine($"Runtime directory: {Path.GetDirectoryName(typeof(object).Assembly.Location)}");
+                }
+            }
+            """;
+
+            if (searchLocation != null)
+                testProject.AdditionalProperties.Add("AppHostDotNetSearch", searchLocation);
+
+            if (appRelativeDotNet != null)
+                testProject.AdditionalProperties.Add("AppHostRelativeDotNet", appRelativeDotNet);
+
+            // Identifer based on test inputs to create test assets that are unique for each test case
+            string assetIdentifier = $"{searchLocation}{appRelativeDotNet}{expectedLocation}";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: assetIdentifier);
+
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand.Execute()
+                .Should().Pass();
+
+            // Published apphost should have .NET search location options changed
+            var publishDirectory = publishCommand.GetOutputDirectory(targetFramework: targetFramework).FullName;
+            TestCommand runCommand = new RunExeCommand(Log, Path.Combine(publishDirectory, $"{testProject.Name}{Constants.ExeSuffix}"));
+
+            string expectedRoot = null;
+            switch (expectedLocation)
+            {
+                case "AppRelative":
+                    // Copy the host and runtime to the expected .NET root
+                    expectedRoot = Path.Combine(publishDirectory, appRelativeDotNet);
+                    CopyDirectory(Path.Combine(TestContext.Current.ToolsetUnderTest.DotNetRoot, "host"), Path.Combine(expectedRoot, "host"));
+                    CopyDirectory(Path.Combine(TestContext.Current.ToolsetUnderTest.DotNetRoot, "shared", "Microsoft.NETCore.App"), Path.Combine(expectedRoot, "shared", "Microsoft.NETCore.App"));
+                    break;
+                case "EnvironmentVariable":
+                    // Set DOTNET_ROOT environment variable to the expected .NET root
+                    expectedRoot = TestContext.Current.ToolsetUnderTest.DotNetRoot;
+                    runCommand = runCommand.WithEnvironmentVariable("DOTNET_ROOT", expectedRoot);
+                    break;
+                default:
+                    // Should fail - make sure DOTNET_ROOT is not set
+                    runCommand = runCommand.WithEnvironmentVariable("DOTNET_ROOT", string.Empty);
+                    break;
+            }
+
+            var result = runCommand.Execute();
+            if (expectedRoot != null)
+            {
+                // On macOS, /tmp actually points to /private/tmp - the app will print the resolved path
+                if (OperatingSystem.IsMacOS() && expectedRoot.StartsWith("/tmp/"))
+                {
+                    expectedRoot = $"/private{expectedRoot}";
```

Comment:
Is this just to workaround symlink equivalence? Couldn't we use a .NET API for that?